### ISSUE TITLE
Styling the div/iframe

### ISSUE THREAD

**Issue by andytate1968** [microsoft/powerbi-client-angular#20]
<powerbi-report [embedConfig]="reportConfig" 
     [phasedEmbedding]="phasedEmbeddingFlag" [eventHandlers]="eventHandlersMap">
</powerbi-report>

How can I set the height and width of the iFrame that is produced? Also, how can I get the report to align to the left of the iFrame? Currently it shows a 150px height iFrame. I want to also be able to set the border to none. I know you can pass a css class to it but can't find what you can set using that i.e. what is the format of it?

**Comment by LuisCarlosRajoy** [comment]
Hi, I've been facing this same issue and found a "css workaround" in my case I'm using Angular 12 , in my template's main styles css file, I've addded the next line:
powerbi-report > div {
  height: (your hieght)px!important;
}
Quick note, I've tried in my component stylesheet file first, and didn't work.

**Comment by KotanaSai21** [comment]
Refer to [<powerbi-report> component and <iframe> rendering with fixed height instead of full screen · Issue #25 · microsoft/powerbi-client-angular (github.com)](https://github.com/microsoft/powerbi-client-angular/issues/25) issue for understanding in brief.

Keeping the discussion in detail, Follow the steps to apply styling to div/iframe

1. Pass the cssClassName as input to the PowerBi report component. 

```
<powerbi-report [embedConfig]="reportConfig" cssClassName="report-container"
[phasedEmbedding]="phasedEmbeddingFlag" [eventHandlers]="eventHandlersMap">
```

2. This class name is assigned to the div where the iframe is embedded. Apply styling to div in `styles.css` file

```
.report-container {
  height: 75vh;
  margin: 8px auto;
  width: 90%;
}
```

Applying the styling in `app.component.css` is not reflecting because powerbi-report component is nested. If you want to apply styling in app.component.css use `::ng-deep` to apply stylings or set encapsulation to `ViewEncapsulation.None`.

Feel free to reach out for any other queries.
